### PR TITLE
[WIP] Filters List

### PR DESCRIPTION
**Pull request by jeremiehelme**
Arranging the filters list so that it's more usable, and adapted to mobile. Hoping that it will be used more

filters-desktop

filters-tablet

filters-mobile

### CODE REVIEW

**Comment by SamMousa** [comment]
Why do these changes require a structural change in the HTML?
This should be possible with pure CSS, right?

**Comment by SamMousa** [comment]
Also, where do the numbers go in the smaller screen? 
This is the current mobile view: 
image

**Comment by jeremiehelme** [comment]
The numbers are just above, I just took the screenshot a little lower :

filters-mobile-numbers

**Comment by SamMousa** [comment]
Then these questions remain: 
Why do these changes require a structural change in the HTML?
This should be possible with pure CSS, right?

My goal is to have clean and semantic HTML and not add block elements just for layout purposes.

**Comment by jeremiehelme** [comment]
About the structural changes, I added two divs :
- one to display the input list with it's own scrollbar
- one to align the numbers correctly

File: protected/views/project/view/filters.php [hunk 1]
```diff
@@ -20,47 +20,49 @@
...
+    JS
+            )
 
-    ?>
+        ?>
     <div class="advanced">
         <div class="filter filter_search">
             <?= Icon::search(); ?>
-            <input id="search-filter">
+            <input id="search-filter" placeholder="Search">
```
Comment (SamMousa) [inline protected/views/project/view/filters.php]: This is the only change in this file as far as I can tell. Revert the change to indenting please.

File: public/css/dashboard.css [hunk 2]
```diff
@@ -209,82 +231,121 @@ span.header {
     background-color: white;
     grid-area: filters;
     margin-bottom: 0;
-    padding: 10px;
-    font-size: 20px;
+    padding: 15px 0 10px;
+    font-size: 18px;
+    display: block;
```
Comment (SamMousa) [inline public/css/dashboard.css]: This is a div, it's default display mode is already block?

File: public/css/dashboard.css [hunk 3]
```diff
@@ -93,20 +93,42 @@ a {
     /*display:none;*/
 }
 
+.filter .flatpickr-input[readonly] {
```
Comment (SamMousa) [inline public/css/dashboard.css]: I don't think this is related to flatpickr (the datepicker component) or the fact that it marks the input as read only.

The targeted selector should (please confirm) probably just be something like `.filter input[type=text]` or even `.filter input`.

File: public/css/dashboard.css [hunk 4]
```diff
@@ -93,20 +93,42 @@ a {
...
+    color: #999;
+    }
+
 .filter_search {
-    border-radius: 15px;
-    padding: 10px;
-    margin-bottom: 20px;
+    padding: 20px 10px 20px 10px;
+    margin: 0 0 20px 0;
+    width: 100%;
```
Comment (SamMousa) [inline public/css/dashboard.css]: Having an `inline-block` AND `width: 100%` is the same as the default `display: block`, right?

**Comment by SamMousa** [comment]
When I look at your changes locally I see an icon that seems to have been lost along the way:
image


Should I remove that icon or does it need to be visible somewhere?

**Comment by SamMousa** [comment]
I'm getting strange behavior on Chrome, when expanding the filters on it's a small window the length goes to like 4000px even though when I scroll down I get no content.

Can you confirm?

**Comment by jeremiehelme** [comment]
I moved the search Icon back inside the search input, but had to use an extra div.
The other option would be to use an image as background for the input, but we wouldn't be able to use the icon

**Comment by jeremiehelme** [comment]
> I'm getting strange behavior on Chrome, when expanding the filters on it's a small window the length goes to like 4000px even though when I scroll down I get no content.
> 
> Can you confirm?

Yes. Same on safari. I'll have a look at it

**Comment by jeremiehelme** [comment]
For some reason, the grid doesn't understand the overflow hidden, and act like if the filters were still taking the whole height

**Comment by jeremiehelme** [comment]
I finally positionned the list in absolute. It works for the different screen sizes and don't add extra space to the bottom anymore

**Pull request merged**